### PR TITLE
v2 further work

### PR DESCRIPTION
**Pull request by steelbrain**
Docs checklist
- [ ] Introduction
  - [x] Getting started
  - [ ] CLI Usage
  - [ ] Configuration
  - [x] Components
  - [x] Presets
  - [ ] ...
- [ ] Technical
  - [x] Lifecycles
  - [x] Components
  - [x] Presets
  - [ ] ...

### CODE REVIEW

File: docs/technical/components.md [hunk 1]
```diff
@@ -13,9 +13,9 @@ It is important to note that all callback methods of the component are invoked w
 
 Another important thing to note is that all callback methods receive the user-config merged with default config as their first parameter. The given config object is cloned for every invocation so you cannot use it to store state of your component.
 
-All components regardless of their types have two lifecycle methods, `activate()` and `dispose()`. They are invoked when a component is added to Pundle and when Pundle is disposed or the component is removed respectively.
+All components regardless of their types have atleast two lifecycle methods, `activate()` and `dispose()`. They are invoked when a component is added to Pundle and when Pundle is disposed or the component is removed respectively.
```
Comment (natew) [inline docs/technical/components.md]: missing space between `at least`

File: docs/technical/components.md [hunk 2]
```diff
@@ -117,9 +117,9 @@ module.exports = createPlugin(function(config, file) {
 
 ### Reporter Components
 
-Reporter Components are used to output the reports of Pundle core or other components. It has a main callback that recieves an error or any `Issue` constructed by `pundle-api` package.
+Reporter Components are used to output the reports of Pundle and it's components. It has a main callback that recieves an error or any `Issue` constructed by `pundle-api` package.
 
-Reporter Components are where all of `pundle.compilation.report(...)` results goto. Each report is fed to all of registered reporter components. Each of them can have different means of output. For example, the default reporter writes the errors and other output to CLI, you could write one to output to websockets etc.
+Reporter Components are where all of `pundle.compilation.report(...)` input goes to. Each report is fed to all of registered reporter components. They can have different means of output. For example, the default reporter writes the errors and other output to CLI, you could write one to output to websockets etc.
```
Comment (natew) [inline docs/technical/components.md]: you can just write `input goes` btw, but `input goes to` works

File: docs/technical/components.md [hunk 3]
```diff
@@ -174,7 +174,7 @@ module.exports = createPostTransformer(function(config, contents) {
 
 ## Watcher Components
 
-Watcher Components are Pundle components that have more than one main callback. They are used when Pundle is running in watcher or dev server mode. Watcher Components usually have `tick`, `ready` and `compile` callbacks.
+Watcher Components are Pundle components that have more than one main callback. They are used when Pundle is running in watcher or dev server mode. Watcher Components have `tick`, `ready` and `compile` callbacks, all of which are optional; though you must at least specify one of them.
```
Comment (natew) [inline docs/technical/components.md]: In this context parens is more appropriate I think:

```
, all of which are optional (though you must at least specify one of them).
```

**Comment by steelbrain** [comment]
I wish we could keep this PR focused but Pundle is still in development so this PR is not so much focused as docs aren't the main priority right now, merging as is, will still keep releasing betas from master

**Pull request merged**